### PR TITLE
Impl user follows

### LINKED ISSUE #54: Follows

### PR DESCRIPTION
**Pull request by mmmommm**
close #54

### CODE REVIEW

File: retweets.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func postRetweet(ctx context.Context, c *client, uid string, tid string) (*PostR
 	if err != nil {
 		return nil, fmt.Errorf("post retweet new request with ctx: %w", err)
 	}
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.bearerToken))
+	req.Header.Set("Authorization", fmt.Sprintf("OAuth %s", c.bearerToken))
```
Comment (mmmommm) [inline retweets.go:78]: I've mixed up the correction of a mistake in a previous retweet pr.
F.Y.I https://developer.twitter.com/en/docs/twitter-api/tweets/retweets/api-reference/post-users-id-retweets

File: client.go [hunk 2]
```diff
@@ -20,7 +20,10 @@ const (
 type Client interface {
 	// CountsFullArchiveTweet(ctx context.Context, query string, opt ...*TweetCountsOption) (*TweetCountsResponse, error)
 	CountsRecentTweet(ctx context.Context, query string, opt ...*TweetCountsOption) (*TweetCountsResponse, error)
-	DeleteRetweet(ctx context.Context, id string, stid string) (*DeleteRetweetResponse, error)
+	// DeleteFollowing(ctx context.Context, suid string, tuid string) (*DeleteFollowingResponse, error)
```
Comment (sivchari) [inline client.go:23]:
```suggestion
	// UndoFollowing(ctx context.Context, suid string, tuid string) (*DeleteFollowingResponse, error)
```

File: client.go [hunk 3]
```diff
@@ -70,10 +74,22 @@ func (c *client) CountsRecentTweet(ctx context.Context, query string, opt ...*Tw
 	return countsRecentTweet(ctx, c, query, opt...)
 }
 
+func (c *client) DeleteFollowing(ctx context.Context, suid string, tuid string) (*DeleteFollowingResponse, error) {
```
Comment (sivchari) [inline client.go:77]:
```suggestion
func (c *client) UndoFollowing(ctx context.Context, suid string, tuid string) (*DeleteFollowingResponse, error) {
```

File: client.go [hunk 4]
```diff
@@ -20,7 +20,10 @@ const (
 type Client interface {
 	// CountsFullArchiveTweet(ctx context.Context, query string, opt ...*TweetCountsOption) (*TweetCountsResponse, error)
 	CountsRecentTweet(ctx context.Context, query string, opt ...*TweetCountsOption) (*TweetCountsResponse, error)
-	DeleteRetweet(ctx context.Context, id string, stid string) (*DeleteRetweetResponse, error)
+	// DeleteFollowing(ctx context.Context, suid string, tuid string) (*DeleteFollowingResponse, error)
+	// DeleteRetweet(ctx context.Context, id string, stid string) (*DeleteRetweetResponse, error)
```
Comment (sivchari) [inline client.go:24]:
```suggestion
	// UndoRetweet(ctx context.Context, id string, stid string) (*DeleteRetweetResponse, error)
```

File: client.go [hunk 5]
```diff
@@ -70,10 +74,22 @@ func (c *client) CountsRecentTweet(ctx context.Context, query string, opt ...*Tw
 	return countsRecentTweet(ctx, c, query, opt...)
 }
 
+func (c *client) DeleteFollowing(ctx context.Context, suid string, tuid string) (*DeleteFollowingResponse, error) {
+	return deleteFollowing(ctx, c, suid, tuid)
+}
+
 func (c *client) DeleteRetweet(ctx context.Context, id string, stid string) (*DeleteRetweetResponse, error) {
```
Comment (sivchari) [inline client.go:81]:
```suggestion
func (c *client) UndoRetweet(ctx context.Context, id string, stid string) (*DeleteRetweetResponse, error) {
```

**Review by sivchari — CHANGES_REQUESTED** [review]

**Review by sivchari — APPROVED** [review]
LGTM

**Pull request merged**